ISSUER_PROTOCOL.md

```diff
@@ -0,0 +1,343 @@
+
+
+#### Redemption Metadata
+
+At redemption time, the metadata encoded in the Trust Token will be embedded in the Signed Redemption Record. The public metadata is embedded via the key ID of the corresponding key used to sign the Trust Token. The private metadata bit is embedded as a one-bit boolean that can be set per redeemer logic and which can be decoded by partners downstream. One potential means of encoding the private metadata is to hash a shared secret and the token hash and take the lowest bit XOR between that and the metadata value (H(shared\_secret||token-hash) ^ private metadata value), allowing only partners with the shared secret to get the true value of the private metadata bit. Other potential schemes involve an additional endpoint which can be used to read the private metadata value based on the Signed Redemption Record (SRR) or token hash.
+
+
+## Trust Token Crypto Protocol
+
+The Trust Token crypto protocol is based on the PMBTokens design in https://eprint.iacr.org/2020/072 (appendix H) using P-384. The necessary keys and function mappings are described below.
+
+
+### Keys
+
+The Trust Token protocol primarily requires two kinds of keypairs:
+
+*   SRRSigningKey/SRRVerificationKey - A Ed25519 public keypair for signing and verifying the integrity of the Signed Redemption Record response.
+*   TrustTokenSecretKey/TrustTokenPublicKey - A Trust Token keypair used to sign and verify Trust Tokens.
+
+#### SRRSigningKey/SRRVerificationKey
+
+These keys should be generated and stored as standard Ed25519 keys, with the public/verification key being included in the Key Commitment result as a 32-byte RFC8032 encoding.
+
+
+#### TrustTokenSecretKey/TrustTokenPublicKey
+
+These are keys used in Trust Token consisting of elliptic curve scalars and points. All scalars and points are sized based on the curve choice (P-384). Up to 3 keys may be configured in parallel in the key commitment.
+
+Each keypair consist of the following:
+
+
+```
+struct {
+  // Corresponding to the FALSE private metadata bit (x0, y0).
+  Scalar x0;
+  Scalar y0;
+  // Corresponding to the TRUE private metadata bit (x1, y1).
+  Scalar x1;
+  Scalar y1;
+  // Corresponding to the token validity (x˜, y˜).
+  Scalar xs;
+  Scalar ys;
+} TrustTokenSecretKey;
+
+struct {
+  ECPoint pub0; // Corresponding to the FALSE private metadata bit.
+  ECPoint pub1; // Corresponding to the TRUE private metadata bit.
+  ECPoint pubs; // Corresponding to the token validity check.
+} TrustTokenPublicKey;
+```
+
+The TrustTokenPublicKey is encoded in the key commitment and should be serialized as follows:
+
+
+```
+struct {
+  opaque pub0<1..2^16-1>; // X9.62 Uncompressed point.
+  opaque pub1<1..2^16-1>; // X9.62 Uncompressed point.
+  opaque pubs<1..2^16-1>; // X9.62 Uncompressed point.
+} TrustTokenPublicKey;
+```
```

Comment:
Looks like TrustTokenPublicKey is being defined twice in a row. (Here and just above.)

Reply:
Done.